cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/logging/Log.java

```diff
@@ -0,0 +1,125 @@
+package cloud.foundry.cli.crosscutting.logging;
+
+import java.util.logging.Level;
+
+/**
+ * The central logging module, providing a standardized interface for all logging activities in the project.
+ * By using this module instead of using custom logging facilities (like System.{out,err}.print*, for instance), a
+ * consistent style is ensured and log messages can be filtered properly.
+ * The interface is similar to Python's logging module, which provides a simple-to-use yet powerful API.
+ */
+public class Log {
+    // we can internally use the Java logging facilities
+    private static final java.util.logging.Logger logger;
+
+    // initialize and configure logger initially
+    static {
+        logger = java.util.logging.Logger.getLogger("cfctl");
+
+        // by default, we don't want to log debug messages
+        // however, the user can opt-in to them by setting the environment variable $DEBUG
+        if (System.getenv("DEBUG") != null) {
+            logger.setLevel(Level.FINE);
+        } else {
+            logger.setLevel(Level.INFO);
+        }
+    }
+
+    /**
+     * Internal helper building one string from multiple objects using a StringBuilder.
+     *
+     * @param arg0 mandatory log argument
+     * @param args optional additional arguments
+     */
+    private static String buildString(Object arg0, Object... args) {
+        StringBuilder message = new StringBuilder();
+
+        /*
+         requiring an initial arg to be passed and then allowing an array of args has two advantages:
+         - first of all, we ensure that we receive at least one argument
+         - second, we can easily ensure there's no unnecessary trailing whitespace, as we can append it before every
+           additional argument, as demonstrated in the loop below
+         */
+
+        message.append(arg0);
+
+        for (Object arg : args) {
+            message.append(" ");
+            message.append(arg);
```

Comment:
Maybe you could try to make the append in one line

Reply:
I was experimenting with different output options, therefore moving these lines up and down. I can also use a single line.